src/test/software/amazon/event/ruler/ACMachineTest.java

```diff
@@ -1277,6 +1277,29 @@ public void testAnythingButDeletion() throws Exception {
         assertTrue(cut.isEmpty());
     }
 
+    @Test
+    public void testAnythingButSuffix() throws Exception {
+
+        String rule = "{\n" +
```

Comment:
How's the test coverage?  Normally when we're extending the pattern syntax I'd like to see the tests explore possible new syntax errors and also combinations, e.g.
```json
{"a": [ {"prefix": "foo"}, {"anything-but": "footer"} ] }
```

Reply:
thanks Tim.  I'll model the prefix tests, also found a bug I need to squash.  Will re-open the PR when the fix is ready

Reply:
thanks @timbray please let me know what more is needed!

Reply:
Tests look much better.  Heh, I had missed the fact that anything-but/suffix hadn't actually been implemented.  I'll let others who are closer to the current state of the machine-building comment on the implementation, but FWIW I didn't see anything that looked really wrong.

Reply:
Thank you... and indeed I think I initially misinterpreted @baldawar 's comments on #66, but in any event I think this is a probably a decent PR and doesn't appear to have performance regressions.